### PR TITLE
added functionality of pressed button

### PR DESCRIPTION
**Pull request by LucreciaLucero**
Created folders for mvp, contract and added button functionality

image
image

### CODE REVIEW

File: app/src/main/java/com/example/parkingsystemjava/MainActivity.java [hunk 1]
```diff
@@ -2,11 +2,26 @@
...
 import android.os.Bundle;
 
+import com.example.parkingsystemjava.databinding.ActivityMainBinding;
+import com.example.parkingsystemjava.mvp.model.ParkingModel;
+import com.example.parkingsystemjava.mvp.presenter.ParkingPresenter;
+import com.example.parkingsystemjava.mvp.view.ParkingView;
+
 public class MainActivity extends AppCompatActivity {
+    private ActivityMainBinding binding;
+    private ParkingPresenter presenter;
```
Comment (mauroLonghi) [inline app/src/main/java/com/example/parkingsystemjava/MainActivity.java]: Here you need to use the contract type, and then when you instantiate the atributte yo use the clases

File: app/src/main/java/com/example/parkingsystemjava/mvp/view/ParkingActivityView.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package com.example.parkingsystemjava.mvp.view;
+
+import android.app.Activity;
+import android.app.FragmentManager;
+import android.content.Context;
+import androidx.annotation.Nullable;
+import java.lang.ref.WeakReference;
+
+public class ParkingActivityView {
```
Comment (lucianoeli) [inline app/src/main/java/com/example/parkingsystemjava/mvp/view/ParkingActivityView.java]: I think this name could be renamed...
this view its going to be the parent  for all your future  activity views. So i think it should be called BaseActivityView, or ActivityView. 

its not to bad, but i think you might confuse with ParkingView

File: app/src/main/java/com/example/parkingsystemjava/mvp/model/ParkingModel.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package com.example.parkingsystemjava.mvp.model;
+
+import com.example.parkingsystemjava.mvp.contract.ParkingContract;
+
+public class ParkingModel implements ParkingContract.ModelContract {
+
+    private int parkingAvailables = 0;
```
Comment (lucianoeli) [inline app/src/main/java/com/example/parkingsystemjava/mvp/model/ParkingModel.java:7]: its good to have these values in constants

File: app/src/main/java/com/example/parkingsystemjava/mvp/presenter/ParkingPresenter.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
+package com.example.parkingsystemjava.mvp.presenter;
+
+import com.example.parkingsystemjava.mvp.contract.ParkingContract;
+
+public class ParkingPresenter implements ParkingContract.PresenterContract{
+
+    private ParkingContract.ModelContract model;
+    private ParkingContract.ViewContract view;
+
+    public ParkingPresenter(ParkingContract.ModelContract model,ParkingContract.ViewContract view){
```
Comment (lucianoeli) [inline app/src/main/java/com/example/parkingsystemjava/mvp/presenter/ParkingPresenter.java]: you have some identations problems here.
remember to format (ctrl + alt + l ) every file you touch

File: app/src/main/java/com/example/parkingsystemjava/MainActivity.java (hunk 1 above)
Comment (LucreciaLucero) [reply app/src/main/java/com/example/parkingsystemjava/MainActivity.java]: done

File: app/src/main/java/com/example/parkingsystemjava/mvp/presenter/ParkingPresenter.java (hunk 4 above)
Comment (LucreciaLucero) [reply app/src/main/java/com/example/parkingsystemjava/mvp/presenter/ParkingPresenter.java]: thanks! done

File: app/src/main/java/com/example/parkingsystemjava/mvp/view/ParkingActivityView.java (hunk 2 above)
Comment (LucreciaLucero) [reply app/src/main/java/com/example/parkingsystemjava/mvp/view/ParkingActivityView.java]: done, changed the name from "ParkingActivityView" to "ActivityView".

**Review by mauroLonghi — APPROVED** [review]

File: app/src/main/java/com/example/parkingsystemjava/mvp/contract/ParkingContract.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+    interface ModelContract{
+        void setParkingAvailables();
+        int getParkingAvailables();
+    }
+
+    interface ViewContract{
+        void showParkingsAvailable(int parkingsAvailable);
+    }
+
```
Comment (SartiMau) [inline app/src/main/java/com/example/parkingsystemjava/mvp/contract/ParkingContract.java]: remove this extra space pls

File: app/src/main/java/com/example/parkingsystemjava/mvp/view/ActivityView.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
+package com.example.parkingsystemjava.mvp.view;
```
Comment (SartiMau) [inline app/src/main/java/com/example/parkingsystemjava/mvp/view/ActivityView.java]: this should be in a base package inside the view package

File: app/src/main/java/com/example/parkingsystemjava/mvp/view/ParkingView.java [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+    public ParkingView(Activity activity, ActivityMainBinding binding) {
+        super(activity);
+        this.binding = binding;
+    }
+
+    @Override
+    public void showParkingsAvailable(int parkingsAvailable) {
+        Context context = getContext();
+        Toast.makeText(context, context.getString(R.string.toast_main_parking_size, parkingsAvailable), Toast.LENGTH_LONG).show();
```
Comment (SartiMau) [inline app/src/main/java/com/example/parkingsystemjava/mvp/view/ParkingView.java]: please, check if the context is not null before use it

File: app/src/main/java/com/example/parkingsystemjava/mvp/view/ParkingView.java [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+    }
+
```
Comment (SartiMau) [inline app/src/main/java/com/example/parkingsystemjava/mvp/view/ParkingView.java]: remove this extra space

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/parkingsystemjava/mvp/view/ParkingView.java (hunk 7 above)
Comment (LucreciaLucero) [reply app/src/main/java/com/example/parkingsystemjava/mvp/view/ParkingView.java]: done

File: app/src/main/java/com/example/parkingsystemjava/mvp/contract/ParkingContract.java (hunk 5 above)
Comment (LucreciaLucero) [reply app/src/main/java/com/example/parkingsystemjava/mvp/contract/ParkingContract.java]: done

File: app/src/main/java/com/example/parkingsystemjava/mvp/view/ActivityView.java (hunk 6 above)
Comment (LucreciaLucero) [reply app/src/main/java/com/example/parkingsystemjava/mvp/view/ActivityView.java]: done

File: app/src/main/java/com/example/parkingsystemjava/mvp/view/ParkingView.java (hunk 8 above)
Comment (LucreciaLucero) [reply app/src/main/java/com/example/parkingsystemjava/mvp/view/ParkingView.java]: done

**Review by SartiMau — COMMENTED** [review]
Rememeber that all files should end with an empty line

**Review by nicolbidone — APPROVED** [review]

**Review by SartiMau — APPROVED** [review]

**Review by BQuinteros — APPROVED** [review]

**Review by lucianoeli — APPROVED** [review]

**Pull request merged**